src/com/stuypulse/stuylib/math/interpolation/NearestInterpolator.java

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2022 StuyPulse Robotics. All rights reserved. */
+/* This work is licensed under the terms of the MIT license */
+/* found in the root directory of this project. */
+
+package com.stuypulse.stuylib.math.interpolation;
+
+import com.stuypulse.stuylib.math.SLMath;
+import com.stuypulse.stuylib.math.Vector2D;
+
+/**
+ * This class is a modified version of Linear Interpolation (Lerp). Using this class will allow you to interpolate
+ * between the NEAREST two reference points, as opposed to just interpolating between only two points in Linear Interpolator.
```

Comment:
remove the first sentence. probably want to make the comment just:

Using this class will allow you to interpolate between the two points NEAREST to an input, as opposed to just interpolating between only two points.